File: benchmarks/beginner/B1-fibonacci/go/main.go
package main

import (
	"fmt"
	"math"
	"runtime"
	"time"
)

func fib(n int) int {
	if n <= 1 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	const N = 40
	const RUNS = 10
	timings := make([]float64, RUNS)

	fmt.Println("=== BENCHMARK: B1 - Fibonacci ===")
	fmt.Println("Language: Go")
	fmt.Printf("Runtime:  Go %s\n", runtime.Version())
	fmt.Println()

	var result int
	for i := 0; i < RUNS; i++ {
		start := time.Now()
		result = fib(N)
		elapsed := float64(time.Since(start).Microseconds()) / 1000.0 // ms
		timings[i] = elapsed
		fmt.Printf("Run %2d: %.2f ms\n", i+1, elapsed)
	}

	fmt.Println()
	minT := timings[0]
	maxT := timings[0]
	sum := 0.0
	for _, t := range timings {
		sum += t
		if t < minT {
			minT = t
		}
		if t > maxT {
			maxT = t
		}
	}
	avg := sum / float64(RUNS)
	variance := 0.0
	for _, t := range timings {
		diff := t - avg
		variance += diff * diff
	}
	stddev := math.Sqrt(variance / float64(RUNS))

	fmt.Printf("Min:    %.2f ms\n", minT)
	fmt.Printf("Avg:    %.2f ms\n", avg)
	fmt.Printf("Max:    %.2f ms\n", maxT)
	fmt.Printf("StdDev: %.2f ms\n", stddev)
	fmt.Println()
	status := "✗"
	if result == 102334155 {
		status = "✓"
	}
	fmt.Printf("Verification: %d %s\n", result, status)
}

File: benchmarks/beginner/B1-fibonacci/python/main.py
import time
import math


def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)


def main():
    N = 40
    RUNS = 10
    timings = []

    print("=== BENCHMARK: B1 - Fibonacci ===")
    print("Language: Python")
    print(f"Runtime:  Python {__import__('sys').version.split()[0]}")
    print()

    for i in range(1, RUNS + 1):
        start = time.perf_counter()
        result = fib(N)
        elapsed = (time.perf_counter() - start) * 1000  # ms
        timings.append(elapsed)
        print(f"Run {i:2d}: {elapsed:.2f} ms")

    print()
    min_t = min(timings)
    avg_t = sum(timings) / len(timings)
    max_t = max(timings)
    variance = sum((t - avg_t) ** 2 for t in timings) / len(timings)
    stddev = math.sqrt(variance)

    print(f"Min:    {min_t:.2f} ms")
    print(f"Avg:    {avg_t:.2f} ms")
    print(f"Max:    {max_t:.2f} ms")
    print(f"StdDev: {stddev:.2f} ms")
    print()
    status = "✓" if result == 102334155 else "✗"
    print(f"Verification: {result} {status}")


if __name__ == "__main__":
    main()
